### PR TITLE
Issue 22950 start stop experiment

### LINKED ISSUE #22950: Start/Stop Experiment
Introduce the ability to start and stop an experiment. 

**Start path: `api/v1/experiments/{id}/_start`**

Preconditions to Start:

- [x] Have DRAFT state
- [x] Have at least one variant 
- [x] Have primary goal set 

Scheduling considerations in order to Start:

- [x] If no startTime provided, set startTime to now() 
- [x] if no endTime provided set to 35 days
- [x] Fail if startTime provided is in the past
- [x] Fail if endTime provided is in the past
- [x] Fail if endTime is not after startTime 
- [x] Duration limit is 35 days

**Stop path: `api/v1/experiments/{id}/_end`**

Preconditions to End:

- [x] Have RUNNING state

Additional requirements: 

- [x] Postman tests

### PR DESCRIPTION
**Pull request by dsilvam**
Implementation for Starting and Ending an Experiment from REST. 
Includes postman tests.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java [hunk 1]
```diff
@@ -16,6 +18,8 @@
 
 public interface ExperimentsAPI {
 
+    int EXPERIMENT_MAX_DURATION = Config.getIntProperty("EXPERIMENT_MAX_DURATION", 35);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java:21]: I think should be lazy

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 2]
```diff
@@ -157,6 +164,99 @@ public List<Experiment> list(ExperimentFilter filter, User user) throws DotDataE
...
+        DotPreconditions.checkArgument(UtilMethods.isSet(experimentId), "experiment Id must be provided.");
+
+        final Optional<Experiment> persistedExperimentOpt =  find(experimentId, user);
+
+        DotPreconditions.isTrue(persistedExperimentOpt.isPresent(),()-> "Experiment with provided id not found",
+                DoesNotExistException.class);
+
+        final Experiment experimentFromFactory = persistedExperimentOpt.get();
+        validatePermissions(user, experimentFromFactory,
+                "You don't have permission to archive the Experiment. "
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:181]: should be "...start the experiment..."

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 3]
```diff
@@ -157,6 +164,99 @@ public List<Experiment> list(ExperimentFilter filter, User user) throws DotDataE
...
+                + "have the Goal set.");
+
+        final Experiment experimentToStart;
+
+        if(experimentFromFactory.scheduling().isEmpty()) {
+            final Scheduling scheduling = startNowScheduling(experimentFromFactory);
+            experimentToStart = experimentFromFactory.withScheduling(scheduling);
+        } else {
+            Scheduling scheduling = validateScheduling(experimentFromFactory.scheduling().get());
+            experimentToStart = experimentFromFactory.withScheduling(scheduling);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:205]: do we need to do this? I mean in this case the experiment already has the scheduling set, right?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentForm.java [hunk 4]
```diff
@@ -124,4 +135,39 @@ public ExperimentForm build() {
             return new ExperimentForm(this);
         }
     }
+
+    private void validateScheduling(final Scheduling scheduling) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentForm.java:139]: is this the same? https://github.com/dotCMS/core/pull/22969/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR274

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 5]
```diff
@@ -157,6 +164,99 @@ public List<Experiment> list(ExperimentFilter filter, User user) throws DotDataE
...
+    @Override
+    public Experiment start(String experimentId, User user)
+            throws DotDataException, DotSecurityException {
+        DotPreconditions.isTrue(hasValidLicense(), InvalidLicenseException.class,
+                invalidLicenseMessageSupplier);
+        DotPreconditions.checkArgument(UtilMethods.isSet(experimentId), "experiment Id must be provided.");
+
+        final Optional<Experiment> persistedExperimentOpt =  find(experimentId, user);
+
+        DotPreconditions.isTrue(persistedExperimentOpt.isPresent(),()-> "Experiment with provided id not found",
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:176]: better use elseThrow

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java:21]: Done

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:181]: Done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentForm.java (hunk 4 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentForm.java:139]: Refactored to reuse

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java (hunk 5 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java]: Done

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java [hunk 6]
```diff
@@ -16,6 +19,8 @@
 
 public interface ExperimentsAPI {
 
+    Lazy<Integer> EXPERIMENT_MAX_DURATION = Lazy.of(()->Config.getIntProperty("EXPERIMENT_MAX_DURATION", 35));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPI.java:22]: nice

**Review by oidacra — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 7]
```diff
@@ -157,6 +162,100 @@ public List<Experiment> list(ExperimentFilter filter, User user) throws DotDataE
         return factory.list(filter);
     }
 
+    @Override
+    @WrapInTransaction
+    public Experiment start(String experimentId, User user)
+            throws DotDataException, DotSecurityException {
+        DotPreconditions.isTrue(hasValidLicense(), InvalidLicenseException.class,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:169]: nice

**Pull request merged**